nemo/collections/tts/models/radtts.py

```diff
@@ -0,0 +1,387 @@
+            f0=f0,
+            energy_avg=energy_avg,
+            voiced_mask=voiced_mask,
+            p_voiced=p_voiced,
+        )
+        loss_outputs = self.criterion(outputs, in_lens, out_lens)
+
+        loss = None
+        for k, (v, w) in loss_outputs.items():
+            if w > 0:
+                loss = v * w if loss is None else loss + v * w
+
+        if (
+            binarize
+            and self.train_config.kl_loss_start_iter >= 0
+            and self.global_step >= self.train_config.kl_loss_start_iter
+        ):
+            binarization_loss = self.attention_kl_loss(outputs['attn'], outputs['attn_soft'])
+            loss += binarization_loss
+        else:
+            binarization_loss = torch.zeros_like(loss)
+        loss_outputs['binarization_loss'] = binarization_loss
+
+        return {
+            "loss_outputs": loss_outputs,
+            "attn": outputs["attn"] if batch_idx == 0 else None,
+            "attn_soft": outputs["attn_soft"] if batch_idx == 0 else None,
+            "audiopaths": "audio_1" if batch_idx == 0 else None,
+        }
+
+    def validation_epoch_end(self, outputs):
+
+        loss_outputs = outputs[0]["loss_outputs"]
+
+        for k, v in loss_outputs.items():
+            if k != "binarization_loss":
+                self.log("val/" + k, loss_outputs[k][0])
+
+        attn = outputs[0]["attn"]
+        attn_soft = outputs[0]["attn_soft"]
+
+        self.tb_logger.add_image(
+            'attention_weights_mas',
+            plot_alignment_to_numpy(attn[0, 0].data.cpu().numpy().T, title="audio"),
+            self.global_step,
+            dataformats='HWC',
+        )
+
+        self.tb_logger.add_image(
+            'attention_weights',
+            plot_alignment_to_numpy(attn_soft[0, 0].data.cpu().numpy().T, title="audio"),
+            self.global_step,
+            dataformats='HWC',
+        )
+        self.log_train_images = True
+
+    def configure_optimizers(self):
+        logging.info("Initializing %s optimizer" % (self.optim.name))
+        if len(self.train_config.finetune_layers):
+            for name, param in model.named_parameters():
+                if any([l in name for l in self.train_config.finetune_layers]):  # short list hack
+                    logging.info("Fine-tuning parameter", name)
+                    param.requires_grad = True
+                else:
+                    param.requires_grad = False
+        if self.optim.name == 'Adam':
+            optimizer = torch.optim.Adam(
+                self.model.parameters(), lr=self.optim.lr, weight_decay=self.optim.weight_decay
+            )
+        elif self.optim.name == 'RAdam':  # Fasle for inference riva
+            optimizer = torch.optim.RAdam(
+                self.model.parameters(), lr=self.optim.lr, weight_decay=self.optim.weight_decay
+            )
+        else:
+            logging.info("Unrecognized optimizer %s!" % (self.optim.name))
+            exit(1)
```

Comment:
Probably want this to be `logging.error` if it's going to exit immediately after. I'd suggest raising a `ValueError` (or other error) here so the user also gets something easy to look at when things grind to a halt.